internal/server/data/encryption_key.go

```diff
@@ -1,32 +1,59 @@
 package data
 
 import (
+	"fmt"
 	mathrand "math/rand"
 
-	"gorm.io/gorm"
-
+	"github.com/infrahq/infra/internal/server/data/querybuilder"
 	"github.com/infrahq/infra/internal/server/models"
 )
 
-func CreateEncryptionKey(db GormTxn, key *models.EncryptionKey) (*models.EncryptionKey, error) {
+type encryptionKeysTable models.EncryptionKey
+
+func (e encryptionKeysTable) Table() string {
+	return "encryption_keys"
+}
+
+func (e encryptionKeysTable) Columns() []string {
+	return []string{"algorithm", "created_at", "deleted_at", "encrypted", "id", "key_id", "name", "root_key_id", "updated_at"}
+}
+
+func (e encryptionKeysTable) Values() []any {
+	return []any{e.Algorithm, e.CreatedAt, e.DeletedAt, e.Encrypted, e.ID, e.KeyID, e.Name, e.RootKeyID, e.UpdatedAt}
+}
+
+func (e *encryptionKeysTable) ScanFields() []any {
+	return []any{&e.Algorithm, &e.CreatedAt, &e.DeletedAt, &e.Encrypted, &e.ID, &e.KeyID, &e.Name, &e.RootKeyID, &e.UpdatedAt}
+}
+
+func CreateEncryptionKey(tx WriteTxn, key *models.EncryptionKey) error {
+	switch {
+	case key.Name == "":
+		return fmt.Errorf("a name is required for EncryptionKey")
+	case key.RootKeyID == "":
+		return fmt.Errorf("a root key ID is required for EncryptionKey")
+	case key.Algorithm == "":
+		return fmt.Errorf("an algorithm is required for EncryptionKey")
+	}
 	if key.KeyID == 0 {
 		// not a security issue; just an identifier
 		key.KeyID = mathrand.Int31() // nolint:gosec
 	}
 
-	if err := add(db, key); err != nil {
-		return nil, err
-	}
-
-	return key, nil
-}
-
-func GetEncryptionKey(db GormTxn, selector SelectorFunc) (result *models.EncryptionKey, err error) {
-	return get[models.EncryptionKey](db, selector)
+	return insert(tx, (*encryptionKeysTable)(key))
 }
 
-func ByEncryptionKeyID(keyID int32) SelectorFunc {
-	return func(db *gorm.DB) *gorm.DB {
-		return db.Where("key_id = ?", keyID)
+func GetEncryptionKeyByName(tx ReadTxn, name string) (*models.EncryptionKey, error) {
+	table := &encryptionKeysTable{}
+	query := querybuilder.New("SELECT")
+	query.B(columnsForSelect(table))
+	query.B("FROM encryption_keys")
```

Comment:
I think it's an oversight that we never moved encryption_keys to org-specific. Not much to do about it here.